src/cdomains/unionFind.ml

```diff
@@ -1055,8 +1056,8 @@ module LookupMap = struct
       [Z.zero, t]
     | Some zmap ->
       let offset_term_product (z, term_set) =
-        let terms = TSet.to_list term_set in
-        List.cartesian_product [z] terms
+        let terms = TSet.elements term_set in
+        BatList.cartesian_product [z] terms
```

Comment:
`cartesian_product` with a singleton list? This could just be a normal `map`.

Reply:
I have now simplified it.